core/include/prometheus/histogram.h

```diff
@@ -6,21 +6,52 @@
 #include "prometheus/counter.h"
 
 namespace prometheus {
+
+/// \brief A histogram metric to represent aggregatable distributions of events.
+///
+/// This class represents the metric type histogram:
+/// https://prometheus.io/docs/concepts/metric_types/#histogram
+///
+/// A histogram tracks the number of observations and the sum of the observed
+/// values, allowing to calculate the average of the observed values.
+///
+/// At its core a histogram has a counter per bucket. The sum of observations
+/// also behaves like a counter.
+///
+/// See https://prometheus.io/docs/practices/histograms/ for detailed
+/// explanations of histogram usage and differences to summaries.
 class Histogram {
  public:
   using BucketBoundaries = std::vector<double>;
 
-  static const MetricType metric_type = MetricType::Histogram;
-
-  Histogram(const BucketBoundaries& buckets);
-
-  void Observe(double value);
-
-  ClientMetric Collect();
+  static const MetricType metric_type{MetricType::Histogram};
+
+  /// \brief Create a histogram with manually choosen buckets.
+  ///
+  /// The BucketBoundaries are a list of monotonically increasing values
+  /// representing the bucket boundaries. Each consecutive pair of values is
+  /// interpreted as a half-open interval [b_n, b_n+1) which defines one bucket.
+  ///
+  /// There is no limitation on how the buckets are divided, i.e, equal size,
+  /// exponential etc..
+  ///
+  /// The bucket boundaries cannot be changed once the histogram is created.
+  Histogram(const BucketBoundaries&);
+
+  /// \brief Observe the given amount.
+  ///
+  /// The given amount selects the 'observed' bucket. The observed bucket is
+  /// chosen for which the given amount falls into the half-open interval [b_n,
+  /// b_n+1). The counter of the observed bucket is incremented. Also the total
+  /// sum of all observations is incremented.
+  void Observe(double);
```

Comment:
Is there a reason behind removing the argument names?

Reply:
It is the style currently used in many places of the library in the header file. Personally I would add the argument names also in the header (it also eases documenting with doxygen). I reverted the changes made to the argument names.